webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/facts/nodes/NodeFact.scala

```diff
@@ -1452,18 +1641,70 @@ object QueryContext {
  * - access all or none nodes, whatever properties the node has
  * - access nodes only if they belongs to one of the listed tenants.
  */
-sealed trait NodeSecurityContext { def value: String }
+sealed trait NodeSecurityContext {
+  def value:     String
+  // serialize into a string that can be parsed by parse `NodeSecurityContext.parse`
+  def serialize: String
+}
 object NodeSecurityContext       {
 
   // a context that can see all nodes whatever their security tags
-  case object All                                      extends NodeSecurityContext { override val value = "all"  }
+  case object All                                      extends NodeSecurityContext {
+    override val value = "all"
+    override def serialize: String = "*"
+  }
   // a security context that can't see any node. Very good for performance.
-  case object None                                     extends NodeSecurityContext { override val value = "none" }
+  case object None                                     extends NodeSecurityContext {
+    override val value     = "none"
+    override def serialize = "-"
+  }
   // a security context associated with a list of tenants. If the node share at least one of the
   // tenants, if can be seen. Be careful, it's really just non-empty interesting (so that adding
   // more tag here leads to more nodes, not less).
   final case class ByTenants(tenants: Chunk[TenantId]) extends NodeSecurityContext {
-    override val value: String = s"tags:[${tenants.mkString(", ")}]"
+    override val value: String = s"tags:[${tenants.map(_.value).mkString(", ")}]"
+    override def serialize = tenants.map(_.value).mkString(",")
+  }
+
+  /*
+   * Tenants name are only alphanumeric (mandatory first) + '-' + '_' with two special cases:
+   * - '*' means "all"
+   * - '-' means "none"
+   * None means "all" for compat
+   */
+  def parse(tenantString: Option[String], ignoreMalformed: Boolean = true): PureResult[NodeSecurityContext] = {
+    parseList(tenantString.map(_.split(",").toList.map(_.trim)))
+  }
+
+  def parseList(tenants: Option[List[String]], ignoreMalformed: Boolean = true): PureResult[NodeSecurityContext] = {
+    tenants match {
+      case scala.None => Right(NodeSecurityContext.All) // for compatibility with previous versions
+      case Some(ts)   =>
+        (ts
+          .foldLeft(Right(NodeSecurityContext.ByTenants(Chunk.empty)): PureResult[NodeSecurityContext]) {
+            case (x: Left[RudderError, NodeSecurityContext], _) => x
+            case (Right(t1), t2)                                =>
+              t2.strip() match {
```

Comment:
Maybe we should use `.strip()` starting from now : https://stackoverflow.com/a/51266583

The `.trim` is maybe repetitive here since we already `strip` in the `parseList`